### PR TITLE
#161966897 Implement test cases for password reset and forgot password

### PR DESCRIPTION
**Pull request by rkemmy**
#### What does this PR do?
Add password-reset and forgot password tests.

#### Description of Task to be completed?
* Test password-reset
* Test forgot-password functionality

#### How should this be manually tested?
* `git clone -b test-password-reset-161966897 https://github.com/andela/ah-infinity-stones` and ` cd ah-infinity-stones`.
* Run `python3 -m venv virtual` to create a virtual environment and `source virtual/bin/activate` to activate it.
* `pip install -r requirements.txt` to install dependencies
* Create a test database in the postgres shell
* Set up environment variables using the sample below as a guide:
```export DATABASE_USER='<postgres username>'
export DATABASE_PASSWORD='<postgres user's password>'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='<name_of_test_database>' 
```
* Run `python3 manage.py makemigrations` followed by `python3 manage.py migrate` to apply migrations
* Run `python3 manage.py test` in the terminal to run the tests.

#### What are the relevant pivotal tracker stories?
[#161966897](https://www.pivotaltracker.com/story/show/161966897)

#### Screenshots (if appropriate)
image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -93,3 +95,6 @@ ENV/
 
 # SQLite3
 db.sqlite3
+
+# migrations
+authors/apps/*/migrations/*
```
Comment (r-wambui) [inline .gitignore]: We do not git ignore migrations files.

File: authors/apps/authentication/tests/test_reset_password.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            format='json')
+
+        response = self.client.put(
+            'api/users/forgot_password',
+            data=json.dumps(self.renew_password),
+            format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(json.loads(response.data)["message"],
+                         "New password cannot be the same as the old password")
```
Comment (r-wambui) [inline authors/apps/authentication/tests/test_reset_password.py:85]: Great use of pep 8

File: authors/apps/authentication/tests/test_reset_password.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            'old': 'password1',
+            'new_password': 'Password123'
+        }
+
+        self.old_password = {
+            'email': 'remmy@test.com',
+            'old': 'password1'
+        }
+
+    def test_forgot_password(self):
```
Comment (r-wambui) [inline authors/apps/authentication/tests/test_reset_password.py]: This function name and the docsting are quite conflicting. Forgot password and checking if a user receives reset password are a bit different.

File: .gitignore (hunk 1 above)
Comment (rkemmy) [reply .gitignore]: Working on fixing the migrations errors that arise with that change. Thanks

File: authors/apps/authentication/tests/test_reset_password.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        ''' Test that checks if a user receives a reset password mail'''
+
+        response = self.client.post(
+            'api/users/forgot_password',
```
Comment (codeplus254) [inline authors/apps/authentication/tests/test_reset_password.py:37]: Django did not raise issues with this (and the others) URLs? You literally don't have any modifications in urls.py?

Comment (rkemmy) [reply authors/apps/authentication/tests/test_reset_password.py:37]: No, it did not. I do not have any views yet.

File: authors/apps/authentication/tests/test_reset_password.py (hunk 3 above)
Comment (rkemmy) [reply authors/apps/authentication/tests/test_reset_password.py]: On it.

**Comment by PromasterGuru** [comment]
Thumb up for the well-documented PR. Just check on the relevant pivotal tracker stories section as you work on the other feedbacks. I would recommend using the story id [#161966897](https://www.pivotaltracker.com/story/show/161966897).

File: .gitignore (hunk 1 above)
Comment (rkemmy) [reply .gitignore]: It's all fixed now.
I have removed the migrations folder from .gitignore

File: authors/apps/authentication/tests/test_reset_password.py (hunk 3 above)
Comment (rkemmy) [reply authors/apps/authentication/tests/test_reset_password.py]: I have changed the function name and added a more descriptive docstring

**Pull request closed without merging**